### ISSUE TITLE
Developing ReSharper.FSharp on Linux

### ISSUE THREAD

**Issue by mcon** [JetBrains/fsharp-support#107]
I'm aware this isn't explicitly supported at the moment as per the README, but thought I'd have a quick go to see how far off this is from being possible.

 * Running `./gradlew prepare` ran without blowing up
 * Changing the target frameworks to `netcoreapp3.1`, it seems that quite a few of the referenced packages are netframework only
 * A bunch of winforms references don't look promising for Linux

Is anyone familiar with how far away this support is, what the downstream dependencies are, and whether any help would be appreciated.

**Comment by auduchinok** [comment]
I haven't tried building/running it on Linux but speaking from my experience it's possible to bulid the plugin, run, and debug it loaded to a sandboxed Rider on macOS which should be a similar experience to doing it on Linux.

There's no need to change the target frameworks due to use of the reference assemblies package:
https://github.com/JetBrains/fsharp-support/blob/dbee3578ab7ab1170f55481b5169fcfc1b27b325/ReSharper.FSharp/Directory.Build.props#L19

The tests are quite a blocker when you are working on the code. I was promised we'll be able to run tests soon but there are no estimates for now. Moving to .NET Core can greatly improve things since many related things are updated and rewritten in the platform.
There is a recent workaround in the Unity plugin for that: https://github.com/JetBrains/resharper-unity/pull/1560. It didn't work for me on the first try and I'm going to look inti it again later since I'd be happy not to be forced to use Windows for development most of the time.

Attaching to a Mono runtime sometimes works worse than on .NET Framework in terms of debugging ReSharper SDK parts and I'm hoping moving to .NET Core will significantly improve things here.

TL;DR Everything except running ReSharper tests should work mostly fine on Linux.

If there's something that won't work for you, please let me know so we could try to improve it.

**Comment by auduchinok** [comment]
There was a bug in 2019.3.x releases that would make Rider choose an MSBuild from Mono instead of the Core one on this solution and it wouldn't work. If you're using that release please choose a Core MSBuild in `Preferences | Build, Execution, Deployment | Toolset and Build`.

**Comment by mcon** [comment]
Thanks for getting back to me, so I can get a build working using MSBuild from netcore 3.1.102, but as you suggest I can't run the tests due to (not unsurprising) library load errors.

I did have to make a trivial change to a project file to get the build working however: https://github.com/JetBrains/fsharp-support/pull/108

I was hoping to be able to develop a new code completion feature, but as you say: it's a non-starter without being able to run the tests! Sounds like I should be waiting for you guys to get onto netcore, or digging out an old machine to run Windows on!

**Comment by Sejsel** [comment]
I tested this now, found one generated file that wasn't placed in the correct location (made a PR, #165). Tests are still not supported, but other than that everything seems to work.

**Comment by arialdomartini** [comment]
Update after 4 years: how is the support of Linux at the moment?
I'm getting errors both running `./gradlew prepare` and `dotnet restore & dotnet build`.

**Comment by auduchinok** [comment]
> I'm getting errors both running ./gradlew prepare and dotnet restore & dotnet build.

@arialdomartini Could you share them please? I see no issues running both of these commands on Windows and macOS.
